easy-ui-react/src/ProductLayout/ProductLayoutSidebar.tsx

```diff
@@ -0,0 +1,92 @@
+import React, { ReactNode, useCallback } from "react";
+import { Overlay, useDialog, useModalOverlay } from "react-aria";
+import { classNames } from "../utilities/css";
+import { ScreenSizeRenderer } from "./ScreenSizeRenderer";
+import { useProductLayout } from "./context";
+
+import styles from "./ProductLayoutSidebar.module.scss";
+
+export type ProductLayoutSidebarProps = {
+  /**
+   * The content to render in the sidebar.
+   */
+  children: ReactNode;
+};
+
+export function ProductLayoutSidebar(props: ProductLayoutSidebarProps) {
+  const { sidebarTriggerState } = useProductLayout();
+  const handleScreenSizeChange = useCallback(
+    (isLargeScreen: boolean) => {
+      if (isLargeScreen) {
+        sidebarTriggerState.close();
```

Comment:
close the menu when the screen size changes so it is in a predictable state when resizing